### PR TITLE
Fix some clippy warnings

### PR DESCRIPTION
**Pull request by messense**
Notable changes:

1. Elide lifetime annotations if possible
2. Prefer dereference over transmute

### CODE REVIEW

**Comment by fafhrd91** [comment]
@messense is it possible to disable clips checks for `ffi2` and `ffi3` and then enable it in travis?

**Comment by messense** [comment]
Didn't find a way to disable lints for a module.

https://github.com/Manishearth/rust-clippy#configuration

File: pyo3cls/src/py_class.rs [hunk 1]
```diff
@@ -99,7 +99,7 @@ fn impl_class(cls: &syn::Ident, base: &syn::Ident,
             }
             impl std::convert::AsRef<PyObjectRef> for #cls {
                 fn as_ref(&self) -> &_pyo3::PyObjectRef {
-                    unsafe{std::mem::transmute(self.as_ptr())}
+                    unsafe{&*(self.as_ptr() as *const pyo3::PyObjectRef)}
```
Comment (fafhrd91) [inline pyo3cls/src/py_class.rs]: @messense seems this is wrong

Comment (messense) [inline pyo3cls/src/py_class.rs]: https://github.com/Manishearth/rust-clippy/wiki#transmute_ptr_to_ref

Any test case to verify that this is wrong?

Comment (fafhrd91) [inline pyo3cls/src/py_class.rs]: yes, I see segfault in async-tokio. but original code might have same problem.

Comment (messense) [inline pyo3cls/src/py_class.rs]: Verified that the original code has the same problem on macOS.

Comment (messense) [inline pyo3cls/src/py_class.rs]: https://github.com/japaric/rust-san Might be useful.

Comment (fafhrd91) [inline pyo3cls/src/py_class.rs]: looks interesting, but it wouldn't help. pyo3 memory leaks was logical and specific to python ref counting

**Pull request merged**